tools/swift-api-digester/swift-api-digester.cpp

```diff
@@ -1106,6 +1100,12 @@ case SDKNodeKind::X:                                                           \
 // Recursively construct a node that represents a type, for instance,
 // representing the return value type of a function decl.
 static SDKNode *constructTypeNode(SDKContext &Ctx, Type T) {
+  // For generic function type, build from its canonical version.
+  if (auto GFT = dyn_cast<GenericFunctionType>(T.getPointer())) {
+    if (!GFT->isCanonical())
+      return constructTypeNode(Ctx, GFT->getCanonicalType());
+  }
+
   SDKNode* Root = SDKNodeInitInfo(Ctx, T).createSDKNode(SDKNodeKind::TypeNominal);
 
   if (auto NAT = dyn_cast<NameAliasType>(T.getPointer())) {
```

Comment:
Why not just convert the Swift type to a string and output that instead of duplicating the compiler's type structure?

And why do we need to model type alias types here at all? Introducing or eliminating type aliases in a method signature is not a source breaking change.